jax/_src/numpy/lax_numpy.py

```diff
@@ -4820,19 +4820,27 @@ def compress(condition, a, axis: Optional[int] = None, out=None):
 @_wraps(np.cov)
 def cov(m, y=None, rowvar=True, bias=False, ddof=None, fweights=None,
         aweights=None):
-  if y is not None: raise NotImplementedError(
-    "jax.numpy.cov not implemented for nontrivial y. "
-    "Open a feature request at https://github.com/google/jax/issues !")
-
   m, = _promote_args_inexact("cov", m)
+  if y is not None:
+    y, = _promote_args_inexact("cov", y)
```

Comment:
It would be better to promote both inputs to the same type; I think something like this would work:
```python
  if y is not None:
    m, y = _promote_args_inexact("cov", m, y)
  else:
    m, = _promote_args_inexact("cov", m)
```
You could also put the check on y's shape within this block to make things more succinct.

Reply:
Thanks for your kind review :)
I agree that promoting both m and y make code looks cleaner!